mapeditor/validator.cpp

```diff
@@ -159,9 +156,18 @@ std::list<Validator::Issue> Validator::validate(const CMap * map)
 		}
 
 		//verification of starting towns
-		for(auto & mp : amountOfCastles)
-			if(mp.second == 0)
-				issues.emplace_back(QString(tr("Player %1 doesn't have any starting town")).arg(mp.first), false);
+		for (auto & playerTCounter : amountOfTowns)
+		{
+			if (playerTCounter.second == 0)
+			{
+				// FIXME: heroesNames are empty even though heroes are on the map
+				// if(map->players[playerTCounter.first].heroesNames.empty())
+				if(amountOfHeroes[playerTCounter.first] == 0)
+					issues.emplace_back(QString(tr("Player %1 has no towns and heroes assigned")).arg(playerTCounter.first + 1), true);
+				else
+					issues.emplace_back(QString(tr("Player %1 doesn't have any starting town")).arg(playerTCounter.first + 1), false);
```

Comment:
```suggestion
					issues.emplace_back(tr("Player %1 has no towns and heroes assigned").arg(playerTCounter.first + 1), true);
				else
					issues.emplace_back(tr("Player %1 doesn't have any starting town").arg(playerTCounter.first + 1), false);
```

Reply:
I changed `std::list` to `std::set` in case there would be some duplicated messages.